### PR TITLE
Sql Scripts - removed two redundancies

### PR DESCRIPTION
**Pull request by shayhatsor**
this should also solve #1341. @veikkoeeva, please take a look.

### CODE REVIEW

File: src/OrleansSQLUtils/CreateOrleansTables_SqlServer.sql [hunk 1]
```diff
@@ -106,8 +105,7 @@ BEGIN
 	(
 		[DeploymentId] NVARCHAR(150) NOT NULL, 
 		[Timestamp] DATETIME2(3) NOT NULL DEFAULT GETUTCDATE(), 
-		[Version] BIGINT NOT NULL,		
-		[ETag] ROWVERSION NOT NULL,
+		[Version] BIGINT NOT NULL DEFAULT 0,
```
Comment (gabikliot) [inline src/OrleansSQLUtils/CreateOrleansTables_SqlServer.sql]: Shay, can you please also update the comments in the code? It now says: 
"-- the ETag is ROWVersion in SQL Server 2005 and later whereas in SQL Server 2000 UNIQUEIDENTIFIER is used"

**Comment by veikkoeeva** [comment]
@shayhatsor Looks good to me.

What do you think, should the column `[OrleansMembershipVersionTable].[Version]` be renamed to `VersionEtag` to make it a closer match to the used parameter `@versionEtag` and hence make it more obvious in the code where the parameter fits in?

**Comment by shayhatsor** [comment]
@veikkoeeva, my thoughts:
* I think that `Version` is human readable. I'd rather change the code and name the parameter something like `TargetTableVersion`, the queries will make more sense. because `versionEtag` doesn't really say anything to me.
* I'd like to change `OrleansMembershipTable.ETag` in the `MSSQL` script to a `BIGINT` like in the `MySQL` script.
* In both sql scripts, rename `OrleansMembershipTable.ETag` to `OrleansMembershipTable.Version` - this aligns all "version" fields to the same name.
* The `MSSQL2000` and `MSSQL2005 or higher` distinction. I'm currently looking at the comparison, it's currently mostly the same. Other than `DateTime` and `NTEXT` which are deprecated, I don't think that the differences will make any noticeable difference in performance.  for example, the usage of `ROWVERSION` is probably somewhat more perfomant, but I'd rather have the same queries for all `MSSQL` versions and make them as simple and standard SQL as possible. Therefore, the distinction will only be about the table definitions.
* as you've noted in the past, `OrleansMembershipTable.SuspectingSilos` and `OrleansMembershipTable.SuspectingTimes` should be in their own table. this makes the distinction between `MSSQL2000` and `MSSQL2005 or higher` only about `DateTime` vs `DateTime2`.

**Comment by veikkoeeva** [comment]
@shayhatsor Renaming the tables and some of the things have crossed to my mind too. Not as you describe, but in some way. The intent was to keep the closer to the Azure Table Storage code and version (up to some common error messages) and maintain it somewhat backwards compatible the original version. Your suggestion looks OK to me.

The SQL Server 2000 part could be done also so that it's split to a separate script and it would do some `ALTER TABLE` modifications. I'm not sure how much support should there be for SQL Server 2000 anymore, but at least one company did use it.

Maybe the question about refactoring and specifically refactoring `Suspecting*` would boil down to if the table structure should be refactored further. Ideas:
* Statistics logging, values as numeric values, one value per row, for instance. This looks like to me a really needed thing to efficiently use analytic functions.
* Split the tables further so that invariant, slowly changing data would be further split to other tables (silo name, IP etc.). This would make inserting and reading a bit more complicated and I'm inclined to think it's not worth (but see further). It's mainly an issue of storage space, but it doesn't look like it would be a problem.

About the analytic function, I've written this before, but one could modify the SQL clauses without modifying Orleans so that the writes would be non-destructive, `INSERT`s only and modifying the `SELECT` scripts accordingly (maybe something like `CreateOrleansTables_Debug_SqlServer`). Combined with statistics logging this could make a powerful tool for two purposes:
* Debugging and performance tuning of the deployment. Especially if one further logs grain connections: grain interactions (connections in time, messages sent), platform resource consumption etc.
* Domain knowledge mining, "business knowledge". What does it means when certain grains interact in certain ways?

Relational engines have powerful tools to mine data (especially now that SQL Server supports natively R). Here storage space may start matter.

**Comment by shayhatsor** [comment]
@veikkoeeva, i'm going to implement every bullet in my [previous comment](https://github.com/dotnet/orleans/pull/1360#issuecomment-177466733) except the last. it might be done later and is a separate concern. 
@gabikliot I'd rather not update the script documentation in this PR, because i plan to refactor it soon.

**Comment by shayhatsor** [comment]
@veikkoeeva, about your comment:
> Renaming the tables and some of the things have crossed to my mind too. Not as you describe, but in some way. The intent was to keep the closer to the Azure Table Storage code and version (up to some common error messages) and maintain it somewhat backwards compatible the original version.

I understand the design decision. but currently, azure is the only one that uses an obscure ETag. All other implementation use numbers for versions, these include : ZooKeeper, MySQL, SqlServer and Consul.

**Comment by gabikliot** [comment]
I am OK with that plan. What ever gets us to more consistentcy between SQL and MySQL and less redundancy.

**Comment by shayhatsor** [comment]
cool, let's first merge this one and I'll prepare a PR with what we talked about here.

**Comment by veikkoeeva** [comment]
@shayhatsor Sure. I'm glad you have the time budget to push this forward. And sure, maybe it is now time to shake off the ATS history.

**Pull request merged**